website/src/components/CompareLegend/index.tsx

```diff
@@ -43,13 +46,43 @@ const CompareLegend = ({
       `}
     >
       <Star fill={diffColors.created} />
-      {`${created} new fields`}
+      {`${created} new ${pluralise(created, 'field')}`}
       <Star fill={diffColors.updated} />
-      {`${updated} updated fields`}
+      {`${updated} updated ${pluralise(updated, 'field')}`}
       <Star fill={diffColors.deleted} />
-      {`${deleted} deleted fields`}
+      {`${deleted} deleted ${pluralise(deleted, 'field')}`}
     </div>
   );
 };
 
+const defaultCount = { updated: 0, created: 0, deleted: 0 };
+export const generateComparisonCounts = (schemas: Schema[]) =>
+  schemas.reduce(
+    (dictionaryCount, schema) => {
+      const schemaCount = schema.fields.reduce(
+        (fieldCount, field) => {
+          switch (field.changeType) {
+            case ChangeType.CREATED:
+              fieldCount.created++;
+              break;
+            case ChangeType.DELETED:
+              fieldCount.deleted++;
+              break;
+            case ChangeType.UPDATED:
+              fieldCount.updated++;
+          }
+          return fieldCount;
+        },
+        { ...defaultCount },
```

Comment:
this works great! I just feel like the importance of the cloning to avoid side effect is a little  understated with this syntax visually... maybe a `_.cloneDeep` would make that more explicit :)

I know I would be tempted to change it to just `defaultCount` so this would help to prevent that :P